### ISSUE TITLE
performance: Get a 100% score on LightHouse metrics `/playground`

### ISSUE THREAD

**Issue by tusharmath** [tailcallhq/tailcallhq.github.io#216]
Ensure the release build of the website gets a 100% score on `/playground`

NOTE: Ignore PWA

<img width="1723" alt="Screenshot 2024-04-25 at 9 11 24 AM" src="https://github.com/tailcallhq/tailcallhq.github.io/assets/194482/5c3ac04e-57e1-49c5-83df-34b762bd9f51">

**Comment by neo773** [comment]
/attempt #216

<div id="algora" />

| [Algora profile](https://console.algora.io/@/neo773) | Completed bounties | Tech | Active attempts | Options |
| --- | --- | --- | --- | --- |
| @neo773 | &nbsp;&nbsp;&nbsp;45 tailcallhq bounties <br/> + 68 bounties from 22 projects | <div align="center">TypeScript, Rust, <br />Go & more</div> | <div align="center"></div> | [Cancel attempt](https://console.algora.io/api/bounties/clvep9jjc0002l70c0y5pm3ao/cancel-attempt) |

**Comment by RamGoel** [comment]
/attempt #216


<div id="algora-attempt" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clvep9jjc0002l70c0y5pm3ao/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by ImBIOS** [comment]
In my opinion, without a huge rewrite from Docusaurus to Astro Starlight or Next.js Server Streaming, it's unrealistic.

![image](https://github.com/user-attachments/assets/f88e3ce2-d26e-4a39-a819-5ad84a409ec3)

Docusaurus official docs itself can't even reach 90%.

https://pagespeed.web.dev/analysis/https-docs-astro-build-en-getting-started/lwrh5preir?form_factor=mobile

![image](https://github.com/user-attachments/assets/1cb1f626-4943-4fcb-8256-e29b22c7afd7)
![image](https://github.com/user-attachments/assets/784e1e63-99c4-441a-9ced-a989172933a1)

If you allow us to migrate it to Astro or Next.js, you might get this delivered shortly. But until then... well IDK.

**Comment by alexmartinfr** [comment]
I concur with @ImBIOS.
The current stack need to be replaced to reach a 100% score on Lighthouse, especially on mobile.
TailwindCSS 4+ brings nice performance improvement, but that alone will not be enough to make a big difference.

Interested to give it a spin, as a proof of concept, if you are up for a full website do-over.

**Comment by neo773** [comment]
@tusharmath 
Hi, could you close this issue? It's been stuck in my GitHub assigned list forever, and I would like to get rid of it.

**Comment by martinp09** [comment]
/attempt #216

Implementation plan: I’m taking a route-level isolation approach rather than trying to squeeze a mobile 100 out of the full Docusaurus shell.

The issue discussion already identified the main constraint: getting a Lighthouse 100 on mobile through the current Docusaurus stack is unrealistic without a broader framework migration. Instead of proposing a full site rewrite, I’m isolating just `/playground/` as a lightweight static route and deferring GraphiQL/React until a user provides an endpoint.

Plan:
- serve `/playground/` from `static/playground/index.html` so the measured first load does not pay the Docusaurus global JS/CSS/hydration cost
- keep the page Tailcall-branded, accessible, and SEO-complete
- lazy-load GraphiQL only after user intent (`?u=` deep link or manual endpoint submit)
- verify Lighthouse mobile + desktop against a production build
- smoke-test that GraphiQL still loads and executes a real query after the deferred assets load

**Comment by Blackbear6997** [comment]
/attempt #216

Implementation plan:
- Move the heavy Docusaurus/GraphiQL experience from `/playground` to `/playground/app`.
- Serve `/playground` as a tiny static launcher so the measured route does not pay the Docusaurus shell, global JS, GraphiQL CSS, or hydration cost by default.
- Preserve existing deep links by redirecting `/playground/?u=...` to `/playground/app/?u=...`.
- Keep the full GraphiQL app available after user intent and verify Lighthouse against a production build.

**Comment by jamilahmadzai** [comment]
/attempt #216

Implementation plan: I am trying a narrower repo-native variant of the static launcher approach. `/playground/` will stay a tiny static, no-framework first-load route for Lighthouse, while the existing Docusaurus/GraphiQL app moves to `/playground/app` so the interactive path keeps the project dependencies instead of loading GraphiQL/React from a CDN. I will verify production build, typecheck, formatting for touched files, Lighthouse mobile/desktop for `/playground/`, and a deep-link smoke path.

**Comment by yZangEren** [comment]
/attempt #216

Plan: keep the full GraphiQL playground available, but isolate the Lighthouse target route. I moved the existing GraphiQL experience to `/playground/app/` and made `/playground/` a tiny static launcher generated at build time, then verified the release build with mobile Lighthouse.

**Comment by yZangEren** [comment]
Submitted PR #618: https://github.com/tailcallhq/tailcallhq.github.io/pull/618

Local release-build Lighthouse mobile result for `/playground/`: performance 100, FCP 0.6s, LCP 0.9s, Speed Index 0.6s, TBT 0ms, CLS 0.009, TTI 0.9s, 3 requests, 5 KiB.

**Comment by goodnight-eriri** [comment]
/attempt #216

Hi maintainers, I’m interested in working on this, but I noticed the issue has an assignee. Could you confirm whether this bounty is still open for external contributors?

If yes, I’ll reproduce the release-build Lighthouse mobile score for `/playground`, identify the blockers, optimize them, and include before/after reports in the PR. PWA will be ignored as requested.

**Comment by k4it00** [comment]
/attempt #216

**Plan:** diagnose first, fix second.
1. Build the site exactly as the release build does and run Lighthouse mobile on `/playground` to get the current scores + full opportunity list (that page's widget/JS is likely the score killer).
2. Fix the top offenders in order of Lighthouse impact: render-blocking requests, unused/large JS, font loading, image formats, TBT/LCP contributors specific to the playground page.
3. Prove it: attach before/after Lighthouse HTML reports to the PR, with the release build, not dev mode.
4. Iterate until mobile Performance = 100 on `/playground`, keeping the existing functionality intact.
